File: stellar-insured-contracts/sdk/mobile/react-native/signing.ts
// Offline transaction signing utility for React Native (TypeScript)
import { ethers } from 'ethers';

export async function signTransactionOffline(txData: object, privateKey: string): Promise<string> {
  const wallet = new ethers.Wallet(privateKey);
  const tx = await wallet.signTransaction(txData);
  return tx;
}

File: stellar-insured-contracts/sdk/mobile/flutter/signing.dart
// Offline transaction signing utility for Flutter (Dart)
import 'package:web3dart/web3dart.dart';

Future<String> signTransactionOffline(Map<String, dynamic> txData, String privateKey) async {
  final credentials = EthPrivateKey.fromHex(privateKey);
  final tx = Transaction(
    to: EthereumAddress.fromHex(txData['to']),
    value: EtherAmount.inWei(BigInt.parse(txData['value'])),
    data: txData['data'] != null ? hexToBytes(txData['data']) : null,
    gasPrice: txData['gasPrice'] != null ? EtherAmount.inWei(BigInt.parse(txData['gasPrice'])) : null,
    maxGas: txData['gasLimit'],
  );
  final signed = await credentials.signTransaction(tx);
  return bytesToHex(signed, include0x: true);
}
